### ISSUE TITLE
Cenos run service httpd start [FAILED] :/etc/httpd/conf.d/wsgi.conf: Cannot load /usr/lib64/httpd/modules/mod_wsgi.so into server: libpython3.5m.so.1.0:

### ISSUE THREAD

**Issue by GitEasonXu** [GrahamDumpleton/mod_wsgi#348]
## Platform env:
- CentOS release 6.9 (Final)
- Anaconda3
- Python3.5
- Httpd :Server version: Apache/2.2.15 (Unix)  Server built:   Jun 19 2018 15:45:13

## Install Step:
- 1: `yum -y install httpd`
- 2: `yum -y install httpd-devel`
- 3: Build: mod_wsgi(mod_wsgi-4.6.4)
      - `./configure LDFLAGS='-Wl,-rpath=/root/anaconda3/lib' --with-python=/root/anaconda3/bin/python`
      - `LD_RUN_PATH=/root/anaconda3/lib make`
      - `make install` 
- 4:Edit `wsgi.conf` file(path:`/etc/httpd/conf.d`)
      - `LoadModule wsgi_module  /etc/httpd/modules/mod_wsgi.so`
## Finished above, start httpd
- `service httpd start`
But failed, get some information shown below:
```
Starting httpd: httpd: Syntax error on line 221 of /etc/httpd/conf/httpd.conf: Syntax error on line 
1 of /etc/httpd/conf.d/wsgi.conf: Cannot load /usr/lib64/httpd/modules/mod_wsgi.so into server: 
libpython3.5m.so.1.0: cannot open shared object file: No such file or directory
                                                           [FAILED]
```
## Check for possible problems follows:
#101 

- 1: `ldd /etc/httpd/modules/mod_wsgi.so`
```
	linux-vdso.so.1 =>  (0x00007ffeedd67000)
	libpython3.5m.so.1.0 => /root/anaconda3/lib/libpython3.5m.so.1.0 (0x00007fcbd1133000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fcbd0f0a000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fcbd0d06000)
	libutil.so.1 => /lib64/libutil.so.1 (0x00007fcbd0b03000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fcbd08fa000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fcbd0676000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fcbd02e2000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fcbd185d000)
```
- 2: `ldconfig -p | grep libpython*`
```
libpython3.5m.so.1.0 (libc6,x86-64) => /root/anaconda3/lib/libpython3.5m.so.1.0
	libpython3.5m.so (libc6,x86-64) => /root/anaconda3/lib/libpython3.5m.so
	libpython3.so (libc6,x86-64) => /root/anaconda3/lib/libpython3.so
	libpython2.6.so.1.0 (libc6,x86-64) => /usr/lib64/libpython2.6.so.1.0
```
-3: `ls -la libpython*`(path:/root/anaconda3/lib)
```
lrwxrwxrwx. 1 root root       20 Aug  3 11:47 libpython3.5m.so -> libpython3.5m.so.1.0
-rwxr-xr-x. 2 root root 10165073 Jul  3  2016 libpython3.5m.so.1.0
-rwxr-xr-x. 2 root root     5285 Jul  3  2016 libpython3.so
```

## More addition
I use `root` username install `Anaconda`  、`httpd` and build `mod_wsgi`

**Comment by GrahamDumpleton** [comment]
Usually you wouldn't use:

```
LDFLAGS='-Wl,-rpath=/root/anaconda3/lib'
```

It should be sufficient to do:

```
LD_RUN_PATH=/root/anaconda3/lib make
```

If you have done a prior build from mod_wsgi source code in the same directory, ensure that you run a:

```
make distclean
```

before running ``configure``.

If the shared library is still not being found by Apache, the first culprit is likely going to be SELinux. The default profile is likely not going to allow Apache to access under ``/root``.

Independent of SELinux, there is also going to be problems with having Anaconda Python installed under ``/root`` directory, as that directory is not readable to others. So even if SELinux is setup to allow the shared library to be found when Apache starts as ``root``, Python will later fail when initialised, after ``root`` privileges are dropped, as the user Apache runs as at that point will not be able to see into ``/root`` directory.

In short, you should not install Anaconda Python under ``/root``. Install it under ``/usr/local/anaconda3`` instead.

**Comment by GitEasonXu** [comment]
@GrahamDumpleton Thank you for your suggestion to help me solve the problem. I created a new username and install anaconda3 in `/newusername/anaconda3/`。

**Comment by GrahamDumpleton** [comment]
Closing this issue on assumption all is good.
